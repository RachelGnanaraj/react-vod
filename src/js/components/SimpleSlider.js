import React from 'react';
import { Link } from 'react-router';
import Slider from 'react-slick';

import MovieCard from './MovieCard';

import './SimpleSlider.scss'

export default class SimpleSlider extends React.Component {
	constructor(props) {
		super(props);
		this.state =  {
			sliderIndex: 0,
			settings: {
				accessibility: true,
				dots: false,
				focusOnSelect: false,
				infinite: true,
				slideCount: this.props.videoData.length,
				slidesToShow: 5,
				slidesToScroll: 1,
				afterChange: this.handleSliderChange
			}
		};
	}

	componentDidMount() {
		window.addEventListener('keydown', this.handleKeyDown, true);
	}

	componentWillUnmount() {
		window.removeEventListener('keydown', this.handleKeyDown);
	}

	handleSliderChange = sliderIndex => {
		this.setState({ sliderIndex })
	}

	handleKeyDown = e => {
		const { sliderIndex } = this.state
		if (e.keyCode == '37') {
			this.sliderPrev()
		}
		else if (e.keyCode == '38') {
			this.slider.slickGoTo(sliderIndex - 5, false)
			this.setState({
				sliderIndex: sliderIndex - 5 
			})
		}
		else if (e.keyCode == '39') {
			this.sliderNext()
		}
		else if (e.keyCode == '40') {
			this.slider.slickGoTo(sliderIndex + 5, false)
			this.setState({
				sliderIndex: sliderIndex + 5 
			})
		}
	}

	sliderNext() {
		this.slider.slickNext();
	}

	sliderPrev() {
		this.slider.slickPrev();
	}

	render() {
		let videos = this.props.videoData;
		let sliderIndex = this.state.sliderIndex;
		if (!videos || videos.length === 0) {
			return <div>Loading videos ...</div>;
		}

		let slides = videos.map((video, index) => {
			return (
				<div key={index}>
					<MovieCard id={video.id} url={video.url} title={video.title} description={video.description}/>
				</div>
			);
		});
		
		return (
			<div className='simple-slider'>
				<Slider ref={c => (this.slider = c)} {...this.state.settings}>
					{slides}
				</Slider>
			</div>
		);
	}
}
