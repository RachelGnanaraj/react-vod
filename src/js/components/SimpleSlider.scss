.simple-slider:hover {
    .movie-card--not-hover {
        opacity: .5;
    }
}