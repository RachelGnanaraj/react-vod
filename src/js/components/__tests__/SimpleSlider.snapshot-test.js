import React from 'react';
import SimpleSlider from '../SimpleSlider';
import renderer from 'react-test-renderer';

jest.mock('react-slick', () => 'Slider');

const videos = [
    { id: 0, url: '', description: 'test', title: 'test' },
]

test('SimpleSlider render when no videos', () => {
    const component = renderer.create(
        <SimpleSlider videoData={[]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('SimpleSlider render with videos', () => {
    const component = renderer.create(
        <SimpleSlider videoData={videos}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
