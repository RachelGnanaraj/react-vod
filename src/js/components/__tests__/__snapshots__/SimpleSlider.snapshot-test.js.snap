// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleSlider render when no videos 1`] = `
<div>
  Loading videos ...
</div>
`;

exports[`SimpleSlider render with videos 1`] = `
<div
  className="simple-slider"
>
  <Slider
    accessibility={true}
    afterChange={[Function]}
    dots={false}
    focusOnSelect={false}
    infinite={true}
    slideCount={1}
    slidesToScroll={1}
    slidesToShow={5}
  >
    <div>
      <div
        className="thumbnail movie-card movie-card--not-hover"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <video>
          <source
            src=""
            type="video/mp4"
          />
        </video>
        <div
          className="caption"
        >
          <h5>
            test
          </h5>
          <p
            className="card-text movie-description"
          >
            test
          </p>
        </div>
      </div>
    </div>
  </Slider>
</div>
`;
